api/base/serializers.py

```diff
@@ -79,15 +79,15 @@ def to_representation(self, value):
         """
         url = super(JSONAPIHyperlinkedIdentityField, self).to_representation(value)
 
-        if self.meta:
-            meta = {}
-            for key in self.meta:
-                meta[key] = _rapply(self.meta[key], _url_val, obj=value, serializer=self.parent)
-            self.meta = meta
-
-            return {'links': {self.link_type: {'href': url, 'meta': self.meta}}}
-        else:
-            return {'links': {self.link_type: url}}
+        meta = {}
+        for key in self.meta or {}:
+            if key == 'count':
+                show_related_counts = self.context['request'].query_params.get('related_counts', False)
+                if utils.is_falsy(show_related_counts):
+                    continue
+            meta[key] = _rapply(self.meta[key], _url_val, obj=value, serializer=self.parent)
+        self.meta = meta
+        return {'links': {self.link_type: {'href': url, 'meta': self.meta}}}
```

Comment:
@pattisdr 

Do you know offhand why we're using ```self.meta``` here instead of just ```meta```, i.e. why do we need to set ```self.meta = meta```?